lib/src/main/java/com/dynatrace/metric/util/MetricLinePreConfiguration.java

```diff
@@ -224,17 +223,11 @@ private void dimension(
 
       String normalizedKey = key;
       if (DimensionKeyValidator.normalizationRequired(key)) {
-        NormalizationResult normalizationResult = Normalizer.normalizeDimensionKey(key);
-
-        normalizedKey = normalizationResult.getResult();
-        if (normalizationResult.messageType() != NormalizationResult.MessageType.NONE) {
-          Supplier<String> messageSupplier = () -> normalizationResult.getMessage();
-          if (!dimensionKeyNormalizationWarnLogged) {
-            logger.warning(() -> String.format(THROTTLE_INFO_TEMPLATE, messageSupplier.get()));
-            dimensionKeyNormalizationWarnLogged = true;
-          } else {
-            logger.fine(messageSupplier);
-          }
+        NormalizationResult normalizeKeyResult = Normalizer.normalizeDimensionKey(key);
+
+        normalizedKey = normalizeKeyResult.getResult();
+        if (normalizeKeyResult.messageType() != NormalizationResult.MessageType.NONE) {
+          normalizationLogger.logDimensionKeyMessage(CLASS_NAME_FOR_LOGGING, normalizeKeyResult);
```

Comment:
This method asks for metricKey, but here it's passing `CLASS_NAME_FOR_LOGGING`. Is this correct?

Reply:
Yes, because when we create the pre-configuration, we don't yet know about a metric key. The pre-config contains things like Dynatrace-specific dimensions, and we set it up before we create the first metric builder. In order to make that clear in the logs, I put the class name of the pre-config in, instead of the metric key. `CLASS_NAME_FOR_LOGGING` is basically the class simple name in curly braces. It will lead to a log line like
```
[WARNING] [{MetricLinePreConfiguration}] Dimension key normalized from 'default 1' to 'default_1'. Further normalization logs for data of the same type will be logged at debug level. 
```